src/main/java/io/leonis/subra/gui/field/Field.java

```diff
@@ -0,0 +1,63 @@
+package io.leonis.subra.gui.field;
+
+import com.googlecode.lanterna.*;
+import com.googlecode.lanterna.TextColor.RGB;
+import com.googlecode.lanterna.gui2.*;
+import java.awt.Color;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+import lombok.Value;
+import org.nd4j.linalg.api.ndarray.INDArray;
+
+/**
+ * @author Jeroen de Jong
+ */
+@Value
+public class Field extends AbstractComponent<Field> {
+  private final INDArray data;
+  private final List<Function<TerminalPosition, TextColor>> backgrounds;
+  private final List<Function<TerminalPosition, Character>> characters;
+
+  @Override
+  protected ComponentRenderer<Field> createDefaultRenderer() {
+    return new ComponentRenderer<Field>() {
+      @Override
+      public TerminalSize getPreferredSize(final Field component) {
+        return new TerminalSize(component.getData().columns(), component.getData().rows());
+      }
+
+      @Override
+      public void drawComponent(final TextGUIGraphics graphics, final Field component) {
+        IntStream.range(0, component.getData().rows())
+          .forEach(x ->
+            IntStream.range(0, component.getData().columns())
+                .mapToObj(y -> new TerminalPosition(x, y))
+                .forEach(position -> {
+                  final TextColor background = component.getBackgrounds().stream()
+                      .map(b -> b.apply(position))
+                      .filter(Objects::nonNull)
+                      .findFirst()
+                      .orElse(new RGB(0, 0, 0));
+                  graphics.setBackgroundColor(background);
+                  graphics.setForegroundColor(this.visibleTextColor(background));
```

Comment:
why not put these two calls in a `map` after `findFirst()`?

Reply:
Since it doesn't matter. The map is only evaluated once as far as I know. I will test this an come back to you

Reply:
I verified;
```java
    IntStream.range(0, 10)
        .map(i -> { System.out.println(i); return i;})
        .findFirst();
```
Simply prints `0` as expected.

Reply:
i'm sorry, i meant `ifPresent`:

```
component.getBackgrounds().stream()
                      .map(b -> b.apply(position))
                      .filter(Objects::nonNull)
                      .findFirst()
                      .ifPresent(background -> {
                          graphics.setBackgroundColor(background);
                          graphics.setForegroundColor(this.visibleTextColor(background));
                      });
```

Reply:
Otherwise i don't have a default background/foreground color

Reply:
if you need it, shouldn't it be configurable?

Reply:
It can be; but the logic remains the same. Since one point may return values, another may not. If another point doesn't return a specific value it should use the default. But if you only set the default at the beginning as your example suggests; you'll re-use the colors as they were in the previous pass.